### ISSUE TITLE
Graphics adapter not working correctly on Windows 98 (potential regression)

### ISSUE THREAD

**Issue by emxd** [dosbox-staging/dosbox-staging#3945]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

Windows 11

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

On the latest release build (https://github.com/dosbox-staging/dosbox-staging/releases/download/v0.82.0-rc1/dosbox-staging-windows-x64-v0.82.0-rc1.zip), Windows 98 starts with an error message and defaults to the vga display driver (640x480x16). 

On the previous release build (https://github.com/dosbox-staging/dosbox-staging/releases/download/v0.81.2/dosbox-staging-windows-v0.81.2.zip) this does not happen and the display driver works correctly.

![dosbox_gdpfxfPuAt](https://github.com/user-attachments/assets/b2d0f74d-eacb-462e-a31f-ff6cd2593567)

### Steps to reproduce the behaviour.

Explain how to reproduce
1. 
2.
3.

### Your configuration

```ini
https://pastebin.com/PP7P7LCe
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by FeralChild64** [comment]
I can confirm this, I even intended to raise the Issue, too - I just didn’t have time yet to reproduce it wirh some more vanilla configuration.

As a workaround you can try to use SciTechSoft DisplayDoctor 6.53 driver - it’s not perfect, produces visual glitches in some apps, but it works.

**Comment by interloper98** [comment]
> I can confirm this, I even intended to raise the Issue, too - I just didn’t have time yet to reproduce it wirh some more vanilla configuration.
> 
> As a workaround you can try to use SciTechSoft DisplayDoctor 6.53 driver - it’s not perfect, produces visual glitches in some apps, but it works.

@FeralChild64 , are you able to bisect it?

I haven't had any luck trying Windows 98 `SETUP.EXE` with a default config.

```shell
dosbox --noprimaryconf
Z:\> mount c.
Z:\> mount d win98-1st-edition.iso -t cdrom
z:\> d:
D:\> setup.exe
```

![image0001-rendered](https://github.com/user-attachments/assets/67726e20-ba4a-4fdb-9e15-1f762b507f35)

It sits there for minutes apparently running SCANDISK.EXE, but never comes back.

**Comment by Grounded0** [comment]
Let's get this fixed for 0.82.0.

Thanks for reporting it.

**Comment by FeralChild64** [comment]
@interloper98 In order to install Windows 98, you have to create a hard disk image, mount it, and boot the Windows 98 startup floppy.

You can't install the Windows 98 on our emulated DOS, unfortunately.
